src/main/java/Deadline.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Represents a deadline task in the chatbot application.
+ * A deadline task includes a due date in addition to the base task properties.
+ */
+public class Deadline extends Task {
+    protected String by; // The deadline date for the task
+
+    /**
+     * Constructs a new Deadline task with specified name, task number, and deadline date.
+     *
+     * @param name The name or description of the deadline task.
+     * @param taskNo The unique number assigned to the task.
+     * @param by The deadline date for the task.
+     */
+    public Deadline(String name, int taskNo, String by) {
+        super(name, taskNo);
+        this.by = by;
+    }
+
+    /**
+     * Marks this deadline task as completed.
+     * Overrides the markedTask method in the Task class to include deadline-specific details in the confirmation message.
+     */
+    @Override
+    public void markedTask() {
```

Comment:
Do you think it would be better to change the names of these functions to present tense (e.g. markTask)?